### ISSUE TITLE
🐛 'per-file-ignores' option does not work for paths which contain spaces

### ISSUE THREAD

**Issue by s-weigand** [PyCQA/flake8#1520]
## TL;DR

If you came here via google and just want quick fix.
1. Don't use spaces in your paths (replace with `-` or `_`)!
2. If you **really really** have to use spaces, replace all of them with `*` in the patter (`path with/spaces yolo.py`->`path*with/spaces*yolo.py`)

## Issue

Installation, in conda env on windows
```
$ pip install  flake8
```

<details>

<summary>$ flake8 --bug-report</summary>

```json
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.7.6",
    "system": "Windows"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "flake8-darglint",
      "version": "1.5.5"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.8.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.4.0"
    }
  ],
  "version": "4.0.1"
}
```
</details>

Player 4 enteres the [issue ping-pong](https://github.com/s-weigand/flake8-nb/issues/190) 😅

I did some quick debugging and found that the cause of the issue is in `parse_files_to_codes_mapping`

https://github.com/PyCQA/flake8/blob/bff62cdf860a70d05ce84e2b5ee494e7f3fd31af/src/flake8/utils.py#L83

## Example

Given the following config

```ini
[flake8_nb]
max-line-length = 100
per-file-ignores =
    path with/spaces yolo.py:E231
```

It gets tokenized the following way

```python
>>> value_
'\npath with/spaces yolo.py:E231'
>>> ret
[('path', ['E231']), ('with/spaces', ['E231']), ('yolo.py', ['E231'])]
```

While a user who uses spaces in their path would need it to be

```python
>>> ret
[('path with/spaces yolo.py', ['E231'])]
```

I guess the only ways to support files with spaces in the path would be to drop the support for splitting at whitespace, which would be a breaking change or not to break anything add an extra flag, which IMHO would be over the top and is the reason this issue started with a **TL;DR**.

Anyway have a nice evening 🥂 (👋  @MarcoGorelli)

**Comment by asottile** [comment]
yep this is a known / intentional limitation to allow it to be specified on the commandline

**Comment by 1kastner** [comment]
Well, in most CLI tools we can use single or double quotation marks to fix paths with spaces. Would that be a suitable extension to circumvent the limitation?

**Comment by asottile** [comment]
I'd rather not, spaces in filenames are a smell anyway and the complexity to support them does not seem worth it
